packages/sdk-cli/src/models/localRelay/relayInfo.test.ts

```diff
@@ -0,0 +1,183 @@
+    },
+    {
+      readJsonFileValue: { port: '5', pid: 7 },
+      readRelayInfoValue: false,
+      name: 'returns false if port is not a number',
+    },
+    {
+      readJsonFileValue: 'obviously not JSON containing Relay Info',
+      readRelayInfoValue: false,
+      name: 'returns false if no relay info has been obtained',
+    },
+    {
+      readJsonFileValue: { not: 'keys we expect' },
+      readRelayInfoValue: false,
+      name: 'returns false if no relay info has been obtained',
+    },
+  ])('$name', async ({ readJsonFileValue, readRelayInfoValue }) => {
+    jest
+      .spyOn(util, 'readJsonFile')
+      .mockResolvedValueOnce(readJsonFileValue as any);
+
+    await expect(readRelayInfo()).resolves.toEqual(readRelayInfoValue);
+  });
+
+  describe('handles errors', () => {
+    it("doesn't log error if file doesn't exist", async () => {
+      jest
+        .spyOn(util, 'readJsonFile')
+        .mockRejectedValueOnce({ code: 'ENOENT' });
+
+      const consoleErrorSpy = jest.spyOn(console, 'error');
+
+      await expect(readRelayInfo()).resolves.toBe(false);
+      expect(consoleErrorSpy).not.toHaveBeenCalled();
+    });
+
+    it("logs error on any error except ENOENT: file doesn't exist ...", async () => {
+      jest
+        .spyOn(util, 'readJsonFile')
+        .mockRejectedValueOnce({ code: 'random' });
+
+      const consoleErrorSpy = jest.spyOn(console, 'error');
+
+      await expect(readRelayInfo()).resolves.toBe(false);
+      expect(consoleErrorSpy).toHaveBeenCalled();
+    });
+  });
+});
+
+describe('pollRelayInfo', () => {
+  /**
+   * Poll for the relay info in interval of 100ms until either:
+   * - Valid relay info obtained
+   * - Timeout
+   *
+   * Here, we need 3 function calls to obtain valid relay info, and the interval is 100ms –
+   * more than enough time for us to poll relay info and not time out.
+   */
+  it('polls until relay info is obtained', async () => {
+    const relayInfo: RelayInfo = { port: 1, pid: 1 };
+    const readRelayInfo = jest
+      .fn()
+      .mockReturnValueOnce(undefined)
+      .mockReturnValueOnce(false)
+      .mockReturnValueOnce(relayInfo);
+
+    await expect(pollRelayInfo(500, 100, readRelayInfo)).resolves.toEqual(
+      relayInfo,
+    );
+
+    expect(readRelayInfo).toHaveBeenCalledTimes(3);
+  });
+
+  /**
+   * With timeout 3500ms, and interval 1000ms, pollRelayInfo can only make max. 3 poll calls + 1 initial, at t = 0.
+   * We check whether polls are really called in the specified interval and not slower/faster (so exactly 3 times).
+   * setInterval/Timeout/etc. are not guaranteed to call functions at __exactly__ the time specified,
+   * so there is a margin of 500ms (3 x 1000ms + 500ms).
+   */
+  it('polls in regular intervals', async () => {
+    const timeout = 3500;
+    const interval = 1000;
+    const readRelayInfo = jest.fn().mockReturnValue(undefined);
+
+    await pollRelayInfo(timeout, interval, readRelayInfo).catch(() => {});
```

Comment:
You can use `expects().rejects` here.

Reply:
What matcher can I use after `rejects` if all I am expecting is for the function to reject?

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/eee559877e708954b84b673a9b1f52bc66470989

Reply:
`.rejects.toThrow()`